drivers/usb/udc/udc_stm32.c

```diff
@@ -515,9 +515,9 @@ static void udc_stm32_mem_init(const struct device *dev)
 	}
 
 	/* The documentation is not clear at all about RX FiFo size requirement,
-	 * Allocate a minimum of 0x40 words, which seems to work reliably.
+	 * 160 has been selected through trial and error.
 	 */
-	words = MAX(0x40, cfg->ep_mps / 4);
+	words = MAX(160, cfg->ep_mps / 4);
```

Comment:
>On USB HS, the previous lower limit of 64 is too small, the value
160 has been chosen apparently through trial and error.

At a minimum, this value should be selected based on the controller type and capabilities, not "trial and error".

Reply:
See the discussion here: https://github.com/zephyrproject-rtos/zephyr/pull/62530
I've been digging through documentation and related issues from when this change was introduced on the old usb device driver, and i cannot seem to find any reason. This configuration is what worked in the old driver, which has been used a lot in practice. But maybe someone at ST can help settle where this value comes from (@erwango)?

Reply:
Unfortunately @Desvauxm-st left the company. So I don't have more information other than what is available in the PR. @marwaiehm-st maybe ?

Reply:
Unfortunately, we tried to find a satisfactory explanation other than "this value works", but until now, we haven't found an explanation for why the minimum value of FIFO RX needs to be 160.